specification/sql/resource-manager/Microsoft.Sql/preview/2017-10-01-preview/InterfaceEndpointProfiles.json

```diff
@@ -0,0 +1,416 @@
+          },
+          {
+            "$ref": "#/parameters/ApiVersionParameter"
+          }
+        ],
+        "responses": {
+          "200": {
+            "description": "Succesfully retrieved the list of interface endpoint profiles.",
+            "schema": {
+              "$ref": "#/definitions/InterfaceEndpointProfilesListResult"
+            }
+          },
+          "default": {
+            "description": "*** Error Responses: ***\n\n * 404 SubscriptionDoesNotHaveServer - The requested server was not found\n\n * 404 ResourceNotFound - The requested resource was not found."
+          }
+        },
+        "x-ms-pageable": {
+          "nextLinkName": "nextLink"
+        },
+        "x-ms-examples": {
+          "List of interface endpoint profiles on server.": {
+            "$ref": "./examples/InterfaceEndpointProfileListByServer.json"
+          }
+        }
+      }
+    }
+  },
+  "definitions": {
+    "InterfaceEndpointProfileProperties": {
+      "description": "The properties of a interface endpoint profile.",
+      "required": [
+        "virtualNetworkSubnetId"
+      ],
+      "type": "object",
+      "properties": {
+        "virtualNetworkSubnetId": {
+          "description": "The ARM resource id of the virtual network subnet.",
+          "type": "string"
+        },
+        "privateIp": {
+          "description": "The Private ip associated with the interface endpoint profile",
+          "type": "string",
+          "readOnly": true
+        },
+        "state": {
+          "description": "State of the interface endpoint profile.",
+          "enum": [
+            "Initializing",
+            "InProgress",
+            "Ready",
+            "Failed",
+            "Deleting",
+            "Unknown"
+          ],
+          "type": "string",
+          "readOnly": true,
+          "x-ms-enum": {
+            "name": "InterfaceEndpointProfileStateType",
+            "modelAsString": true
+          }
+        }
+      }
+    },
+    "Resource": {
+      "description": "ARM resource.",
+      "type": "object",
+      "properties": {
+        "id": {
+          "description": "Resource ID.",
+          "type": "string",
+          "readOnly": true
+        },
+        "name": {
+          "description": "Resource name.",
+          "type": "string",
+          "readOnly": true
+        },
+        "type": {
+          "description": "Resource type.",
+          "type": "string",
+          "readOnly": true
+        }
+      },
+      "x-ms-azure-resource": true
+    },
+    "ProxyResource": {
+      "description": "ARM proxy resource.",
+      "type": "object",
+      "allOf": [
+        {
+          "$ref": "#/definitions/Resource"
+        }
+      ],
+      "properties": {}
+    },
+    "InterfaceEndpointProfile": {
+      "description": "A interface endpoint profile resource.",
+      "type": "object",
+      "allOf": [
+        {
+          "$ref": "#/definitions/ProxyResource"
```

Comment:
any reason not to refer [ProxyResource](https://github.com/Azure/azure-rest-api-specs/blob/7c857063e4b451c7209aa300f01e72665097eaac/specification/sql/resource-manager/common/v1/types.json#L33) from v1 common definitions? like 

```json
      "allOf": [
        {
          "$ref": "../../../common/v1/types.json#/definitions/ProxyResource"
        }
```